### ISSUE TITLE
Extract Okta and Auth0 specific code to modular folder structure

### ISSUE THREAD

**Issue by andrew-rist-tw** [PublicDataWorks/complaint-manager#280]
Blocked by #278 and #279

### Business Context:
Having added support for Okta, we want to be able to modularize the auth implementation to the point where adding support for another auth engine will be much simpler. What we want is to extract everything into methods/constants that are built with the same interface for both Okta and Auth0 and put them under a separate folder (I'm thinking `src/auth` with separate subfolders for each engine, e.g. `src/auth/okta` and `src/auth/auth0`) and an index.js at `src/auth` that exports the functions/constants for the correct auth engine depending on the configuration, so the app just needs to import from `src/auth` but under the covers `src/auth/index.js` is exporting the right stuff

**As an** auth engine other than Okta and Auth0
**I want** complaint-manager support for me to be implementable as easily as possible by a team member or contributor
**So that** I do not become an impediment for another org to adopt complaint-manager

### In Scope

1. Extract all auth related functionality into functions/constants
2. Tweak functions/constants for Okta and Auth0 so that they have the exact same interface
3. Move functions/constants to separate folder structure

### Out of Scope

1. Implementing any auth engines 🙂
2.

### Open Questions

1.
2.
3.

## Acceptance Criteria

1. Scenario:
   Given
   When
   Then

2. Scenario:
   Given
   When
   Then

### Checklist (should be reviewed before grooming as the BA or Pair are writing the card—NOT reviewed for the first time at grooming)

- [ ] Ensure to include all necessary mockups to this card.
- [ ] Discuss what success/failure user feedback messages would be present, if any occur (front end/back end).
- [ ] Discuss Feature Flag: Where could a development flag go to flag off incomplete work? Do we want to keep the flag long-term?
- [ ] If this card is part of a sequence of cards, discuss if this card should be blocked until the testing of previous cards in the sequence have been finished.
- [ ] Have we added edge cases as ACs that we're considered from a QA perspective
- [ ] Should the new functionalities be offered only to users with certain permissions?
- [ ] How should this functionality differ between different orgs, and if it does differ, what is the hingepoint we should use for that functionality?

**Notes:**
